lib/library_main.rb

```diff
@@ -0,0 +1,42 @@
+require 'yaml'
+require 'date'
+class Library_main
+
+    require 'date'
+    attr_accessor :collection
+
+    def initialize
+      @collection = YAML.load_file('./lib/data.yml')
+    end
+
+    def check_avl (title,author)
+      book = @collection.detect { |obj| obj[:item][:title] == title && obj[:item][:author] == author } 
+      return book[:available]
+    end
+
+    def check_out (title,author)
+      avl = check_avl(title,author)
```

Comment:
avoid variable names like `avl` its better to write out `available` or `available_books` etc